src/module.c

```diff
@@ -9730,10 +9730,30 @@ int RM_CommandFilterArgDelete(RedisModuleCommandFilterCtx *fctx, int pos)
  * with the allocation calls, since sometimes the underlying allocator
  * will allocate more memory.
  */
-size_t RM_MallocSize(void* ptr){
+size_t RM_MallocSize(void* ptr) {
     return zmalloc_size(ptr);
 }
 
+/* Same as RM_MallocSize, except it works on RedisModuleString pointers.
+ */
+size_t RM_MallocSizeString(RedisModuleString* str) {
+    serverAssert(str->type == OBJ_STRING);
+    return objectComputeSize(NULL, str, 0, 0);
+}
+
+/* Same as RM_MallocSize, except it works on RedisModuleDict pointers.
+ * Note that the returned value is only the overhead of the underlying structures,
+ * it does not include the allocation size of the keys and values.
+ */
+size_t RM_MallocSizeDict(RedisModuleDict* dict) {
+    /* See streamRadixTreeMemoryUsage */
+    size_t size = sizeof(RedisModuleDict);
+    size += dict->rax->numnodes * sizeof(raxNode);
+    /* Add a fixed overhead due to the aux data pointer, children, ... */
+    size += dict->rax->numnodes * sizeof(long)*30;
```

Comment:
where did you take that constant from? either give a reference, or full detail on how you got to it.
i suppose that once we have #10152 (@yzhaon FYI) we should use it here, but i guess it's ok to introduce the API, and change it's underlying code later.